docs/use-cases/gh-actions-manifest-generation.md

```diff
@@ -0,0 +1,1272 @@
+
+Flux suggests maintaining the only active copy of the decryption key for a cluster inside of that cluster (though there may be a provision for backups, or some alternate keys permitted to decrypt.) This arrangement makes such use cases significantly more complicated to explain, beyond the scope of this guide.
+
+For those uses though, additional Workflow Actions are provided:
+
+The [Decrypt SOPS Secrets] action may be useful and it is mentioned here, (but no example uses are provided.)
+
+The [Sops Binary Installer] action enables more advanced use cases, like encrypting or re-encrypting secrets.
+
+#### Jsonnet Recap
+
+While much of this type of manipulation could be handled in `Kustomization`'s `postBuild`, via `envsubst`, some configurations are more complicated this way. They can be better handled in CI, where access to additional tools can be provided.
+
+By writing YAML manifests into a Git commit, the same manifests that `Kustomize` directly applies, they can be saved for posterity. Or projected out into a new pull request where they can be reviewed before application, or with the proper safe-guards in place they can be applied immediately through a more direct-driven automation.
+
+With generated YAML that Flux applies in the cluster directly from Git commits, **fui-yoh** - that's GitOps!
+
+### Commit Across Repositories Workflow
+
+Flux will not deploy from pushes on just any branch; GitRepository sources target just one specific branch. Merging to a `staging` branch, for example, can be used to trigger a deployment to a Staging environment.
+
+Manifest generation can be used to solve, broadly, very many problems, such that even with many examples, this guide would never be totally exhaustive.
+
+This is the final example in this guide.
+
+Here we show 🥁 ... how to replicate the original behavior of Flux v1's image automation! 🤯 🎉
+
+You can put this workflow in your application repo, and target it toward your `fleet-infra` repo.
+
+To replicate the nearest approximation of Flux's "deploy latest image" feature of yesteryore, we use push events to do the job, as we hinted was possible in an earlier example. This can be done without Flux v1's redundant and expensive image pull behavior, retrieving build metadata required to order image tags for deployment.
+
+Flux recommends using real version numbers in your image tags, with a canonical ordering.
+
+The alternative is racy and doesn't always guarantee the latest commit will be the one that is deployed, since this behavior depends on the time that each commit is pushed, and even precisely how long the build takes to complete; the difference is fine for Dev environments, but this is not a strategy for Production use cases.
+
+Your app's CI can commit and push YAML manifests (or one manifest for each app) into a separate deploy branch for `Kustomization` to apply. The deploy branch in a separate repository should be a branch to which the CI user is granted write access.
```

Comment:
I need to test this with a more than trivial setup. Who is the CI user in GitHub Workflow?

Across repositories, it is whichever user's Personal Access Token was used. Within a single repo, is it whoever pushed the commit that triggered the workflow? (Are there any pitfalls of this, can you still protect the deploy branch and allow writes to it?)

Probably a question that is out of scope for this example use case document, but some folks might trip over this.